src/common/libtaskmap/taskmap.c

```diff
@@ -314,6 +315,98 @@ struct taskmap *taskmap_decode_json (json_t *o, flux_error_t *errp)
     return map;
 }
 
+static int parse_pmi_block (const char *s, int *nodeid, int *count, int *ppn)
+{
+    char *endptr;
+    errno = 0;
+    *nodeid = strtoul (s, &endptr, 10);
+    if (errno != 0 || *endptr != ',')
+        return -1;
+    s = endptr + 1;
+    *count = strtoul (s, &endptr, 10);
+    if (errno != 0 || *endptr != ',')
+        return -1;
+    s = endptr + 1;
+    *ppn = strtoul (s, &endptr, 10);
+    if (errno != 0 || *endptr != ')')
+        return -1;
+    return 0;
+}
+
+static bool is_empty (const char *s)
+{
+    while (isspace(*s) && s++);
+    return *s == '\0';
+}
+
+struct taskmap *taskmap_decode_pmi (const char *s, flux_error_t *errp)
```

Comment:
That is how I had it at first. However, one of the test vectors from libpmi clique tests ensures that an empty string encodes to essentially "(0,1,1)" I think? Anyway, we can't autodetect PMI_process_mapping _and_ support an empty string (which is an invalid RFC 34 taskmap)

https://github.com/flux-framework/flux-core/blob/68a7ea613939cdfd75f68e72ca63614533055433/src/common/libpmi/test/clique.c#L39-L45

Reply:
Oh didn't catch that.  Could "" just be considered another thing to pass to the pmi decoder?  (e.g. invalid RFC 34 like the the `(vector,...` strings, but valid RFC 13)?

Reply:
I thought of that but it seemed pretty kludgy IMO, so I didn't go there. However, if it seems ok to you to have this kind of heuristic (It may break some existing tests, but no big deal)

However, we may need to stop a moment and think here. An empty PMI_process_mapping is supposed to be treated as if all tasks are in their own clique right? (I may not fully understand how it should be treated). Do we need a "special" taskmap flag for this case, since it is the one case that doesn't inherently encode the total number of tasks, won't work for nodeid > 1, etc?

Reply:
Hm, in fact it appears that I've probably broken the empty PMI_process_mapping decoding, since in this PR it is decoded to an "empty" taskmap, i.e. `[]`, which I don't think would pass the `/* mpich unit test */ tests above. Not sure how I dropped the ball on that.

At this point I'm a bit confused on what an empty PMI_process_mapping is supposed to decode to...